src/LegacyStrings.jl

```diff
@@ -49,15 +49,15 @@ using Compat
     end
 
     if VERSION >= v"0.5.0-"
-        immutable ASCIIString <: DirectIndexString
+        struct ASCIIString <: DirectIndexString
```

Comment:
Presumably this package will be mostly unnecessary as packages move away from 0.4 support, so bumping the requirement to 0.6 might not be worth it at this point. Probably no harm in doing so though. Seems like a good question for @tkelman.